pkg/controller/machineset/controller.go

```diff
@@ -18,23 +18,55 @@ package machineset
 
 import (
 	"fmt"
+	"sync"
+	"time"
+
 	"github.com/golang/glog"
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/builders"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/integer"
+
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 	machineclientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	machinesetclientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 	listers "sigs.k8s.io/cluster-api/pkg/client/listers_generated/cluster/v1alpha1"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 )
 
+const (
+	// When batching machine creates, SlowStartInitialBatchSize is the size of the
+	// initial batch.  The size of each successive batch is twice the size of
+	// the previous batch.  For example, for a value of 1, batch sizes would be
+	// 1, 2, 4, 8, ...  and for a value of 10, batch sizes would be
+	// 10, 20, 40, 80, ...  Setting the value higher means that quota denials
+	// will result in more doomed API calls and associated event spam.  Setting
+	// the value lower will result in more API call round trip periods for
+	// large batches.
+	//
+	// Given a number of machines to start "N":
+	// The number of doomed calls per sync once quota is exceeded is given by:
+	//      min(N,SlowStartInitialBatchSize)
+	// The number of batches is given by:
+	//      1+floor(log_2(ceil(N/SlowStartInitialBatchSize)))
+	SlowStartInitialBatchSize = 1
+	// The number of times we retry updating a ReplicaSet's status.
+	statusUpdateRetries = 1
+)
+
 // controllerKind contains the schema.GroupVersionKind for this controller type.
 var controllerKind = v1alpha1.SchemeGroupVersion.WithKind("MachineSet")
 
 // +controller:group=cluster,version=v1alpha1,kind=MachineSet,resource=machinesets
 type MachineSetControllerImpl struct {
 	builders.DefaultControllerFns
 
+	// kubernetesClient a client that knows hwo to consume Node resources
+	kubernetesClient kubernetes.Interface
+
 	// machineClient a client that knows how to consume Machine resources
 	machineClient machineclientset.Interface
```

Comment:
Looking at the way this is used, it seems to be more of a clusterapi interface since the code is pulling objects other than just machines from this.

Reply:
renamed to clusterAPIClient